HIP/hip-851.md

```diff
@@ -0,0 +1,224 @@
+
+Unfortunately, this fact presents a significant challenge for the maintainers and contributors of the Hedera - SemVer, one of the most popular and well-known versioning schemes ever created, is the wrong choice for the Node Software project.
+
+Fortunately, the right choice already exists which can prevent the Node Software project from needing to define its own custom versioning scheme. Another popular, well-defined versioning scheme has already been defined, and the characteristics of this scheme provide a good fit for the unique characteristics of the Node Software project.
+
+That right choice is to implement a flavor of the Calendar Versioning Schema (CalVer). The implementation of this versioning schema change would be made in such a way to remain compatible with many of the beneficial aspects of the SemVer spec that the Node Software project has standardized around in practice, code, and operations.
+
+## Rationale
+
+The [SemVer 2.0.0 Summary](https://semver.org/#summary) describes the current Node Software scheme succinctly:
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+>> MAJOR version when you make incompatible API changes
+>>
+>> MINOR version when you add functionality in a backward compatible manner
+>>
+>> PATCH version when you make backward compatible bug fixes
+>
+> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+This version format is also often referred to with letters e.g `x.y.z` where
+
+- `x` == **MAJOR**
+- `y` == **MINOR**
+- `z` == **PATCH**
+
+Today, the Node Software uses SemVer with MAJOR Version zero (0.y.z) per [rule 4 of the SemVer 2.0.0 Spec](https://semver.org/#spec-item-4):
+
+> MAJOR version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.
+
+At the time of writing for this proposal, the current version of the Node Software is `v0.45.2`. As per rule 4, MAJOR `x` is never incremented, while MINOR `y` can (and does) indicate both major and minor changes that can represent breaking, backwards-incompatible changes. PATCH `z` is used to indicate and deliver important bug fixes for a given MINOR `y` release, but typically does not indicate any significant feature or compatibility change.
+
+### The problem
+
+This usage of the SemVer spec has been appropriate for the Node Software project as it iterated and matured through its early stages and initial major milestones and goalposts. Today however, this usage of the SemVer spec presents 3 significant issues:
+
+1. It is not true that the Node Software is in the initial development stages, as the project has completed most of the initial milestones, providing a robust and mature feature set for the greater Hedera community to build upon.
```

Comment:
I agree, we passed the "1.0" point a long time ago.

One thing to keep in mind is that the node software has two versions -- the software version, and the HAPI version, and they need not have the same versioning scheme. The HAPI version is used to communicate to users of HAPI the compatibility and maturity of the HAPI itself, while the node software version communicates the version of the node software. I suggest that HAPI is semVer, while the node software is calVer

Reply:
@rbair23 I believe your thoughts on this are already captured in the HIP via 1)a) (the line right under the one you commented on here). At least in terms of stating "the node software project AND its public APIs are both in fact pretty stable". If you think there are any revisions/additions that would help describe this or your thoughts on what should or shouldn't be SemVer or CalVer, I'd appreciate any suggestions you may have